scalafix-cli/src/main/scala/scalafix/internal/interfaces/ScalafixArgumentsImpl.scala

```diff
@@ -200,8 +194,16 @@ final case class ScalafixArgumentsImpl(args: Args = Args.default)
   ): ScalafixArguments =
     copy(args = args.copy(scalacOptions = options.asScala.toList))
 
-  override def withScalaVersion(version: String): ScalafixArguments =
-    copy(args = args.copy(scalaVersion = version))
+  override def withScalaVersion(version: String): ScalafixArguments = {
+    ScalaVersion
+      .from(version) match {
+      case Success(value) => copy(args = args.copy(scalaVersion = value))
+      case Failure(exception) =>
+        throw new ScalafixMainArgsException(
+          s"Failed when parsing Scala Version because: ${exception.getMessage}"
+        )
```

Comment:
why not use the constructor that wraps an exception? The `because` is implied then.
```suggestion
        throw new ScalafixMainArgsException("Failed to parse Scala version", exception)
```